### PR TITLE
Split past and future flights

### PR DESCRIPTION
**Pull request by tbergq**
Next up => add gray overlay over cityImage past flight

### CODE REVIEW

File: app/MMB/src/list/FlightList.js [hunk 1]
```diff
@@ -89,28 +107,45 @@ const styles = StyleSheet.create({
...
+          destinationImageUrl(dimensions: _375x165)
+          type
+          oneWay {
+            ...OneWayFlight_booking
+          }
+          return {
+            ...ReturnFlight_booking
+          }
+          multicity {
+            ...MulticityFlight_booking
```
Comment (mrtnzlml) [inline app/MMB/src/list/FlightList.js]: I am sure you don't like these two identical fragments as well. It, in my opinion, indicates wrong components structure. You should compose it a little bit differently: there should be something like `FlightList` with only one fragment and only one responsibility (render the list). Another component should care about the future and past trips. Working example:

```js
export default class Flights extends React.Component<Props> {
  renderInner = (innerProps: FlightsQueryResponse) => {
    return (
      <ScrollView>
        <Translation id="mmb.my_bookings.future_trips" />
        <FlightList data={innerProps.future} />
        <Translation id="mmb.my_bookings.past_trips" />
        <FlightList data={innerProps.past} />
      </ScrollView>
    );
  };

  render = () => (
    <PrivateApiRenderer
      render={this.renderInner}
      query={graphql`
        query FlightsQuery {
          future: allBookings(only: FUTURE) {
            ...FlightList
          }
          past: allBookings(only: PAST) {
            ...FlightList
          }
        }
      `}
    />
  );
}
```

File: app/MMB/src/list/FlightList.js [hunk 2]
```diff
@@ -31,6 +33,7 @@ export class FlightList extends React.Component<Props> {
             type={type}
             imageUrl={imageUrl}
             booking={idx(item, _ => _.node.oneWay)}
+            isPastFlight={isPastFlight}
```
Comment (mrtnzlml) [inline app/MMB/src/list/FlightList.js]: This is another indicator of odd design - these conditional props affecting internal logic. Just keep it simple and use one component in multiple places... :)

**Comment by tbergq** [comment]
@mrtnzlml please check again

File: app/MMB/src/list/FlightList.js [hunk 3]
```diff
@@ -63,51 +62,39 @@ export class FlightList extends React.Component<Props> {
...
-        {flights.map(this.renderItem)}
-      </ScrollView>
-    );
+    const flights = idx(this.props.data, _ => _.edges) || [];
+
+    if (flights.length === 0) {
+      return null;
+    }
+
+    return <React.Fragment>{flights.map(this.renderItem)}</React.Fragment>;
```
Comment (mrtnzlml) [inline app/MMB/src/list/FlightList.js]: @tbergq Why fragment? Can I remove it?

Comment (tbergq) [reply app/MMB/src/list/FlightList.js]: Yes, you are right, the fragment is abundant.

**Comment by mrtnzlml** [comment]
Integrated, thanks. Looks much better now... 🙂

**Pull request closed without merging**